e2e-tests/development-runtime/cypress/integration/gatsby-script/gatsby-script-inline-scripts.js

```diff
@@ -0,0 +1,195 @@
+import { inlineScript } from "../../../gatsby-script-scripts"
+import { resourceRecord, markRecord } from "../../../gatsby-script-records"
+
+const page = {
+  target: `/gatsby-script-inline-scripts/`,
+  navigation: `/gatsby-script-navigation/`,
+}
+
+const typesOfInlineScripts = [
+  {
+    descriptor: `dangerouslySetInnerHTML`,
+    inlineScriptType: inlineScript.dangerouslySet,
+  },
+  {
+    descriptor: `template literals`,
+    inlineScriptType: inlineScript.templateLiteral,
+  },
+]
+
+/**
+ * Normally we would duplicate the tests so they're flatter and easier to debug,
+ * but since the test count grew and the cases are exactly the same we'll iterate.
+ */
+
+for (const { descriptor, inlineScriptType } of typesOfInlineScripts) {
+  describe(`inline scripts set via ${descriptor}`, () => {
+    describe(`using the post-hydrate strategy`, () => {
+      it(`should execute successfully`, () => {
+        cy.visit(page.target)
+
+        cy.getRecord(
+          `post-hydrate-${inlineScriptType}`,
+          `success`,
+          true
+        ).should(`equal`, `true`)
+      })
+
+      it(`should load after the framework bundle has loaded`, () => {
+        cy.visit(page.target)
+
+        // Assert framework is loaded before inline script is executed
+        cy.getRecord(
+          `post-hydrate-${inlineScriptType}`,
+          markRecord.executeStart
+        ).then(dangerouslySetExecuteStart => {
+          cy.getRecord(`framework`, resourceRecord.responseEnd).should(
+            `be.lessThan`,
+            dangerouslySetExecuteStart
+          )
+        })
+      })
+    })
+
+    describe(`using the idle strategy`, () => {
+      it(`should execute successfully`, () => {
+        cy.visit(page.target)
+
+        cy.getRecord(`idle-${inlineScriptType}`, `success`, true).should(
+          `equal`,
+          `true`
+        )
+      })
+
+      it(`should load after other strategies`, () => {
+        cy.visit(page.target)
+
+        cy.getRecord(`idle-${inlineScriptType}`, markRecord.executeStart).then(
+          dangerouslySetExecuteStart => {
+            cy.getRecord(
+              `post-hydrate-${inlineScriptType}`,
+              markRecord.executeStart
+            ).should(`be.lessThan`, dangerouslySetExecuteStart)
+          }
+        )
+      })
+    })
+
+    describe(`when navigation occurs`, () => {
+      it(`should load only once on initial page load`, () => {
+        cy.visit(page.target)
+
+        cy.get(`table[id=script-mark-records] tbody`)
+          .children()
+          .should(`have.length`, 4)
```

Comment:
This is part assertion and I think part flake potential removal as this will (potentially?) wait for expected amount of scripts to report their status before doing assertion on those individual ones